osquery/tables/system/linux/cpu_time.cpp

```diff
@@ -0,0 +1,91 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <boost/algorithm/string/predicate.hpp>
+#include <boost/algorithm/string/trim.hpp>
+
+#include <osquery/tables.h>
+#include <osquery/filesystem.h>
+
+#include "osquery/core/conversions.h"
+
+namespace osquery {
+namespace tables {
+
+const std::string kProcStat = "/proc/stat";
+
+std::vector<std::string> procFromFile(const std::string &path) {
+    std::string content;
+    std::vector<std::string> proc_lines;
+
+    if (!isReadable(path).ok()) {
+        return proc_lines;
+    }
+
+    if (!readFile(path, content).ok()) {
+        return proc_lines;
+    }
+
+    auto lines = split(content, "\n");
+
+    for (auto &line : lines) {
+        boost::trim(line);
+        if (boost::starts_with(line, "cpu")) {
+            proc_lines.push_back(line);
+        }
+    }
+
+    // Remove first cpu line which doesn't give specific core information.
+    if (proc_lines.size() > 0 && proc_lines.front().size() >= 4 &&
+        proc_lines.front().substr(0, 4).compare("cpu ") == 0) {
+        proc_lines.erase(proc_lines.begin());
+    }
+
+    return proc_lines;
+}
+
+static void genCpuTimeLine(const std::string &line, QueryData &results) {
+    Row r;
+
+    std::vector<std::string> words;
+    words = split(line, " ");
+
+    if (words.size() < 11) {
+        // This probably means there's an error in the /proc/stat file.
+        return;
+    }
+
+    r["core"] = words[0].back();
+    r["user"] = words[1];
+    r["nice"] = words[2];
+    r["system"] = words[3];
+    r["idle"] = words[4];
+    r["iowait"] = words[5];
+    r["irq"] = words[6];
+    r["softirq"] = words[7];
+    r["steal"] = words[8];
+    r["guest"] = words[9];
+    r["guest_nice"] = words[10];
+
+    results.push_back(r);
+}
+
+QueryData genCpuTime(QueryContext &context) {
+    QueryData results;
+
+    auto proc_lines = procFromFile(kProcStat);
+    for (auto line : proc_lines) {
```

Comment:
Unless you plan on modifying `line` within the loop use:
```cpp
for (const auto& line : proc_lines)
```

Then you wont make a copy of each (since you're taking a reference `&`), if you did intend to modify you could drop the `const`.